.github/local-actions/branch-manager/lib/main.ts

```diff
@@ -0,0 +1,160 @@
+import * as core from '@actions/core';
+import {PullRequestValidationConfig} from '../../../../ng-dev/pr/common/validation/validation-config.js';
+import {
+  assertValidPullRequestConfig,
+  PullRequestConfig,
+} from '../../../../ng-dev/pr/config/index.js';
+import {loadAndValidatePullRequest} from '../../../../ng-dev/pr/merge/pull-request.js';
+import {AutosquashMergeStrategy} from '../../../../ng-dev/pr/merge/strategies/autosquash-merge.js';
+import {
+  assertValidGithubConfig,
+  getConfig,
+  GithubConfig,
+  setConfig,
+} from '../../../../ng-dev/utils/config.js';
+import {AuthenticatedGitClient} from '../../../../ng-dev/utils/git/authenticated-git-client.js';
+import {
+  ANGULAR_ROBOT,
+  getAuthTokenFor,
+  revokeActiveInstallationToken,
+} from '../../../../github-actions/utils.js';
+import {MergeConflictsFatalError} from '../../../../ng-dev/pr/merge/failures.js';
+import {chdir} from 'process';
+import {spawnSync} from 'child_process';
+
+interface CommmitStatus {
+  state: 'pending' | 'error' | 'failure' | 'success';
+  description: string;
+}
+
+async function main(repo: {owner: string; repo: string}, token: string, pr: number) {
+  if (isNaN(pr)) {
+    core.setFailed('The provided pr value was not a number');
+    return;
+  }
+
+  // Because we want to perform this check in the targetted repository, we first need to check out the repo
+  // and then move to the directory it is cloned into.
+  chdir('/tmp');
+  console.log(
+    spawnSync('git', [
+      'clone',
+      `https://github.com/${repo.owner}/${repo.repo}.git`,
+      './branch-manager-repo',
+    ]).output.toString(),
+  );
+  chdir('/tmp/branch-manager-repo');
```

Comment:
Why not always use the actual target repository (like with `pull_request_target):

* This will be safer
* In a real merge- the caretaker also acts from the main branch
* We could just use the `actions/checkout` as usually done with GitHub actions

Reply:
The idea is that this is a "service" which the other repos are able to call to perform this check, rather than the action being expressed in every repo.  This allows for us to call it in many different ways, via action directly, via API if we need in the future, etc.

> This will be safer
Since we just want to do a static analysis of the git diffs essentially, I don't know that it will be safer, because we are not actually performing any action on the code that is checked out, so regardless of the code change being analyzed, the changes are not executed.

>In a real merge- the caretaker also acts from the main branch
This will be done from the main branch of the repository in question, however its worth noting the caretaker does not have to merge from the main branch.

>We could just use the actions/checkout as usually done with GitHub actions
actions/checkout is controlled at the repo calling it, rather than the source code of the action itself, so we would have to "manage" this configuration at all of the call sites.

Reply:
To be honest (disclaimer: not feeling super strongly): It sounds a little unnecessary to have this in the dev-infra repo if it could just be a shared single action workflow file per repository (compared to having workflows per repo and one here). Also it would simplify the action, and you could still keep the workflow_dispatch even on a per repository basis, right?

Reply:
The reason I think its worth while is because we have to have a separate workflow that exists in each repo which will inform when to run these checks, allowing us to fan our multiple pr checks to run simultaneously.  Rather than having a complex matrix usage (which also comes with github autocanceling things) to communicate which PRs should be checked.  We can just direct all of them here.

Reply:
Sorry, I just need to understand this a little better. Some questions:

1. I don't fully understand what you want to fan out. If it's multiple validations- couldn't the "action code" not create multiple individual statuses without needing an actual matrix?
2. Is the fan-out worth? (need to likely understand better what you are thinking of).

Reply:
>1
The action code can definitely apply statuses to multiple PRs from one action run, but because we need to perform checks using the file system, it is required to be serial

2>
Whenever code is pushed to the main branch of the repository, we will need to recheck all of the PRs which target that branch, for main for instance this is ~all outstanding PRs.  We would like to parallelize these validation checks, and since it requires actually running git commands in the filesystem, we cannot do them in parallel on one VM.  We instead need to fan them out into different runners.  We can do this using an outputted list from one job as an input for a matrix in another, but its very brittle, and if anything in the matrix fails, all others are cancelled.

Reply:
I see, thanks. The second bullet point cleared this up.